pkg/registry/apis/provisioning/register.go

```diff
@@ -479,9 +489,36 @@ func (b *ProvisioningAPIBuilder) GetOpenAPIDefinitions() common.GetOpenAPIDefini
 	return provisioning.GetOpenAPIDefinitions
 }
 
-func (b *ProvisioningAPIBuilder) GetAPIRoutes() *builder.APIRoutes {
-	// TODO: this is where we could inject a non-k8s managed handler... webhook maybe?
-	return nil
+func (b *ProvisioningAPIBuilder) GetPostStartHooks() (map[string]genericapiserver.PostStartHookFunc, error) {
```

Comment:
> as a PostStartHook and not the job queue?

@MissingRoberto which job queue were you referring to here? in k8s code, controllers that run in the same process as the apiserver are normally instantiated outside of a post start hook and then the post start hooks start them (and the related informers). since the controllers are clients of the apiserver, starting them in the post start hook ensures that the apiserver is running and ready to accept their requests.

i also instantiated the client, informer, and controller in the post start hook because the post start hook context passes the loopback config. that's probably not ideal because we will have multiple informer caches if this pattern spreads to other APIs in grafana, but it's something that can be cleaned up and optimized later.

Reply:
Do you mean [jobs/queue.go](https://github.com/grafana/grafana/blob/grafana-git-ui-sync/pkg/registry/apis/provisioning/jobs/queue.go#L20)?

That is a different resource (Job not Repository) so would need its own controller -- if we go that route.  This PR is setting up the callbacks for repositories, not jobs